src/lang/gettext_string_hash.cpp

```diff
@@ -0,0 +1,141 @@
+//
+// Created by machaviprace on 22.07.21.
+//
+
+#include "gettext_string_hash.hpp"
+#include <string.h>
+
+uint32_t gettext_hash_table::hash_string(const char *key) const {
+    uint32_t hval, g;
+    const char *str = key;
+
+    /* Compute the hash value for the given string.  */
+    hval = 0;
+    while (*str != '\0') {
+        hval <<= 4;
+        hval += (unsigned char)*str++;
+        g = hval & ((unsigned long int)0xf << (HASHWORDBITS - 4));
```

Comment:
beware of unsigned long int (depends on the platform!) - please use bit-width specific types like uint32_t